### DISCUSSION TITLE
Prisma Client Connection Closing

### DISCUSSION THREAD

**Discussion by jenningsmax** [prisma/prisma#5665] [Q&A]
I'm setting Prisma up in one of my projects and am running into the following error when PrismaClient tries querying the DB:

```
PrismaClientInitializationError2 [PrismaClientInitializationError]: 
Invalid `prisma.user.findUnique()` invocation:


  Error querying the database: connection closed
    at PrismaClientFetcher.request (/Users/maxjennings/local/lobbby/lobbby/node_modules/@prisma/client/runtime/index.js:78469:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  clientVersion: '2.16.1'
}
```

I'm using Next JS so I have PrismaClient being setup as suggested in docs:

```ts
// prisma.js 
import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
    prisma = new PrismaClient();
} else {
    if (!global.prisma) {
        global.prisma = new PrismaClient();
    }
    prisma = global.prisma;
}
export default prisma;
```

The error is thrown when I try to access it:
```ts
import prisma from "../../../../lib/prisma";

export default async function handle(req, res) {
    const userID = req.query.id;

    const user = await prisma.user.findUnique({
        where: { id: Number(userID) },
    });

    res.status(200).json(user);
}
```

My schema file:
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DB_CONNECTION_STRING")
}

model spatial_ref_sys {
  srid      Int     @id
  auth_name String?
  auth_srid Int?
  srtext    String?
  proj4text String?
}

model User {
  id          Int      @id @default(autoincrement())
  email       String   @unique
  name        String?
  date_joined DateTime @default(now())
  stripe_id   String?
  zoom_token  String?
  active      Boolean
}
```

My db is hosted on Render. I don't believe the issue is coming from the db as I was able to use it successfully with one of the prisma next js example code bases.

Any help is massively appreciated!

**Comment by nikolasburk** [comment]
Hey @jenningsmax 👋  thanks a lot for raising this! This is strange, using the approach to instantiate `PrismaClient` as a singleton should resolve these kinds of issues 🤔 could you [reopen the same question as an actual issue](https://github.com/prisma/prisma/issues/new/choose) so that our Engineering team can pick it up?

**Reply by jenningsmax** [reply]
Turns out I had @prisma/client installed as a dev dependency - this was causing the error

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
